### DISCUSSION TITLE
vsgCs help-The following error occurs after updating. Is it because the graphics card does not support it?

### DISCUSSION THREAD

**Discussion by xingfu2005** [vsg-dev/VulkanSceneGraph#1220] [General]
Warning: miniCompile did recording!
info: Context::waitForCompletion()  00000238BDE49A60 fence->wait() failed with error. VkResult = -4
info: Context::waitForCompletion()  00000238BDE49A60 fence->wait() failed with error. VkResult = -4
info: Context::waitForCompletion()  00000238BDE49A60 fence->wait() failed with error. VkResult = -4
info: Context::waitForCompletion()  00000238BDE49A60 fence->wait() failed with error. VkResult = -4

**Comment by robertosfield** [comment]
This could be a range of reasons, but we can't begin to speculate as you haven't provide any details about your OS/graphics hardware/drivers etc, or what happened before these errors occurred.

**Reply by xingfu2005** [reply]
Thanks.

OS: win10
device:NVIDIA GeForce MX150
"name": "worldviewer.exe
     "args": [
       "--world-file vsgCs/tests/csterrain.json"
     ]
The window is blue and the Earth does not appear.

**Reply by timoore** [reply]
> Thanks.
> 
> OS: win10 device:NVIDIA GeForce MX150 "name": "worldviewer.exe "args": [ "--world-file vsgCs/tests/csterrain.json" ] The window is blue and the Earth does not appear.

I don't see an --ion-token-file argument? The test cases in `tests/` require a Cesium ion token.

**Reply by xingfu2005** [reply]
For convenience，I set ionAccessToken to my Cesium ion token in RuntimeEnvironment.cpp's initializeCs function.

**Reply by xingfu2005** [reply]
It worked fine before untill I update recently

**Reply by xingfu2005** [reply]
2024/1/4:running success in my device.
2014/5/16:
Successfully built using the latest vsgCs,vsg and the latest cesium-native.but the  error is displayed when running

**Comment by robertosfield** [comment]
I haven't used a Geforce MX150 but I guess it should work with Vulkan, a quick search on the web didn't provide any conclusive clarification of this though.

Do the standard vsgExamples like vsgdraw or vsgviewer work? 

To test paging capability you can use the vsgExamples/data/models/openstreetmap.vsgt example:

~~~ sh
vsgviewer models/openstreetmap.vsgt
~~~

Once you have confirmed that the basic Vulkan/VulkanSceneGraph side is work then investigate vsgCs. There is a vsgCs Issue thracker you could use if there are vsgCs specific issues you are having,

**Reply by xingfu2005** [reply]
vsgCs worked last year.
vsgtiledatabase.exe "--osgearth --bing-maps worlked well
I guess new version need some extension my device not support，such as mesh

**Reply by xingfu2005** [reply]
vsgmeshshader.exe --barycentric not work

**Reply by robertosfield** [reply]
OK, then the basic Vulkan/VulkanSceneGraph is working fine, issues with vsgCs is something to bring up with Tim on the vsgCs issue tracker.

vsgmeshshader example requires hardware and drivers that support mesh shaders so if it doesn't work then this is likely just down to your hardware/drivers not having support for mesh shader extensions.

To see what your hardware is capable of try running vsgdeviceselection:

~~~ sh
vsgdeviceselection --extensions
~~~

**Reply by xingfu2005** [reply]
Anyway thanks for the reply！

**Reply by xingfu2005** [reply]
**I add debug option,the failure info like this:**

info: Selected sRGB window 0.0289912 0.0289912 0.133209 1
Warning: miniCompile did recording!
VUID-vkCmdBindDescriptorSets-pDescriptorSets-00358(ERROR / SPEC): msgNum: -507995293 - Validation Error: [ VUID-vkCmdBindDescriptorSets-pDescriptorSets-00358 ] Object 0: handle = 0x1f2a343c520, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xb097c90000000027, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xe1b89b63 | vkCmdBindDescriptorSets(): pDescriptorSets[0] (VkDescriptorSet 0xb097c90000000027[]) being bound is not compatible with overlapping descriptorSetLayout at index 0 of VkPipelineLayout 0x2e2cd000000002b[] due to: VkDescriptorSetLayout 0x3fbcd60000000028[] from pipeline layout has 3 total descriptors, but VkDescriptorSetLayout 0xcb1c7c000000001b[], which is bound, has 5 total descriptors.. The Vulkan spec states: Each element of pDescriptorSets must have been allocated with a VkDescriptorSetLayout that matches (is the same as, or identically defined as) the VkDescriptorSetLayout at set n in layout, where n is the sum of firstSet and the index into pDescriptorSets (https://vulkan.lunarg.com/doc/view/1.3.283.0/windows/1.3-extensions/vkspec.html#VUID-vkCmdBindDescriptorSets-pDescriptorSets-00358)
    Objects: 2
        [0] 0x1f2a343c520, type: 6, name: NULL
        [1] 0xb097c90000000027, type: 23, name: NULL
VUID-vkCmdDrawIndexed-None-08600(ERROR / SPEC): msgNum: 941228658 - Validation Error: [ VUID-vkCmdDrawIndexed-None-08600 ] Object 0: handle = 0x1f2a343c520, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0x421a0f0000000074, type = VK_OBJECT_TYPE_PIPELINE; Object 2: handle = 0xb097c90000000027, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; Object 3: handle = 0x44695a0000000071, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x381a0272 | vkCmdDrawIndexed():  VkDescriptorSet 0xb097c90000000027[] bound as set #0 is not compatible with overlapping VkPipelineLayout 0x44695a0000000071[] due to: VkDescriptorSetLayout 0x8f1c31000000006d[] from pipeline layout has 3 total descriptors, but VkDescriptorSetLayout 0xcb1c7c000000001b[], which is bound, has 5 total descriptors. The Vulkan spec states: For each set n that is statically used by a bound shader, a descriptor set must have been bound to n at the same pipeline bind point, with a VkPipelineLayout that is compatible for set n, with the VkPipelineLayout used to create the current VkPipeline or the VkDescriptorSetLayout array used to create the current VkShaderEXT , as described in Pipeline Layout Compatibility (https://vulkan.lunarg.com/doc/view/1.3.283.0/windows/1.3-extensions/vkspec.html#VUID-vkCmdDrawIndexed-None-08600)
    Objects: 4
        [0] 0x1f2a343c520, type: 6, name: NULL
        [1] 0x421a0f0000000074, type: 19, name: NULL
        [2] 0xb097c90000000027, type: 23, name: NULL
        [3] 0x44695a0000000071, type: 17, name: NULL
VUID-vkCmdDrawIndexed-None-08600(ERROR / SPEC): msgNum: 941228658 - Validation Error: [ VUID-vkCmdDrawIndexed-None-08600 ] Object 0: handle = 0x1f2a343c520, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0x421a0f0000000074, type = VK_OBJECT_TYPE_PIPELINE; Object 2: handle = 0xb097c90000000027, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; Object 3: handle = 0x44695a0000000071, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x381a0272 | vkCmdDrawIndexed():  VkDescriptorSet 0xb097c90000000027[] bound as set #0 is not compatible with overlapping VkPipelineLayout 0x44695a0000000071[] due to: VkDescriptorSetLayout 0x8f1c31000000006d[] from pipeline layout has 3 total descriptors, but VkDescriptorSetLayout 0xcb1c7c000000001b[], which is bound, has 5 total descriptors. The Vulkan spec states: For each set n that is statically used by a bound shader, a descriptor set must have been bound to n at the same pipeline bind point, with a VkPipelineLayout that is compatible for set n, with the VkPipelineLayout used to create the current VkPipeline or the VkDescriptorSetLayout array used to create the current VkShaderEXT , as described in Pipeline Layout Compatibility (https://vulkan.lunarg.com/doc/view/1.3.283.0/windows/1.3-extensions/vkspec.html#VUID-vkCmdDrawIndexed-None-08600)

**Comment by timoore** [comment]
Thanks for the additional details. I'll take a look at this.

**Comment by timoore** [comment]
I'm not seeing these validation errors with the latest VSG, vsgXchange, vsgImGui, and vsgCS. Are you sure that you have the latest versions of all of these (including ImGui; might need to do a `git submodule update --recursive`) and have installed them so that vsgCs is using these latest versions?

I can't really say if the validation errors are related to things not working on the MX150, but it's possible.

**Reply by xingfu2005** [reply]
I returned vsg to 1.1.0, vsgCs and cesium native use the latest version,Just modify wviewer and CsViewer back to compatibility with vsg1.1.0 as before, it ran successfully, and the earth came out.
Has anyone successfully run vsg1.1.4+vsgCs on Windows 10?

**Comment by xingfu2005** [comment]
The reason why the new vsg version of the earth cannot be displayed is probably related to vsg’s changes to ViewDependentState-related shaders and resource descriptors  binding . Modify the vgsCs's VIEW_DESCRIPTOR_SET binding related code in pbr.cpp and replace  with vsgExamples . It's not because of the graphics card.It may also be necessary to modify the shader files, but I'm not familiar with it yet.
